dubbo-cluster/src/test/java/org/apache/dubbo/rpc/cluster/StickyTest.java

```diff
@@ -68,6 +69,8 @@ public void setUp() throws Exception {
         invokers.add(invoker2);
 
         clusterinvoker = new StickyClusterInvoker<StickyTest>(dic);
+
+        ExtensionLoader.resetExtensionLoader(LoadBalance.class);
```

Comment:
I am not convinced either. Why we need to reset extension loader here?

Reply:
When either testStickyForceCheck or testStickyNoCheck runs, they eventually end up using an instance of RoundRobinLoadBalance, which is obtained and shared between the two tests through the ExtensionLoader, accessed using the LoadBalance.class as key. In between the two runs, one of the shared RoundRobinLoadBalance instance's internal map fields changes. The change is such that if testStickyNoCheck runs after testStickyForceCheck, the assertion for testStickyNoCheck fails. The goal of resetting is to make sure the two tests do not end up using the same RoundRobinLoadBalance instance when they run.

I would be happy to discuss any other ways of approaching this problem.

Reply:
I think I understand what happens now, agree with your change.